### PR TITLE
Parsing work for dotnet nuget sign command

### LINKED ISSUE #18219: dotnet nuget sign command
We (NuGet.Client) team completed development of NuGet side changes for dotnet nuget sign command. This issue is created to track SDK changes for this commands.

Here are the links to spec, issue, and PR for NuGet work.

Spec - https://github.com/NuGet/Home/blob/dev/proposed/archive/DotNetSignCommands.md
Issue - https://github.com/NuGet/Home/issues/10581
PR - https://github.com/NuGet/NuGet.Client/pull/3974

### PR DESCRIPTION
**Pull request by heng-liu**
Fixes: https://github.com/dotnet/sdk/issues/18219

Description

I work on NuGet Client team. We added dotnet nuget sign command to dotnet sdk. Added this command to the parser for tab completion.  NuGet.Client completed issue is https://github.com/NuGet/Home/issues/10581

### CODE REVIEW

**Review by wli3 — APPROVED** [review]

File: src/Tests/dotnet.Tests/CommandTests/CompleteCommandTests.cs [hunk 1]
```diff
@@ -275,5 +276,35 @@ public void GivenNuGetTrustCommandItDisplaysCompletions()
...
+                "--certificate-path",
+                "--certificate-store-name",
+                "--certificate-store-location",
+                "--certificate-subject-name",
+                "--certificate-fingerprint",
+                "--certificate-password",
+                "--hash-algorithm",
+                "--timestamper",
+                "--timestamp-hash-algorithm",
+                "--verbosity",
```
Comment (erdembayar) [inline src/Tests/dotnet.Tests/CommandTests/CompleteCommandTests.cs:294]: By looking at existing code it looks we need to order by alphabetic order.
`--output` here.

File: src/Tests/dotnet.Tests/CommandTests/CompleteCommandTests.cs [hunk 2]
```diff
@@ -275,5 +276,35 @@ public void GivenNuGetTrustCommandItDisplaysCompletions()
...
+        [Fact]
+        public void GivenNuGetSignCommandItDisplaysCompletions()
+        {
+            var expected = new[] {
+                "--output",
+                "--certificate-path",
+                "--certificate-store-name",
+                "--certificate-store-location",
+                "--certificate-subject-name",
+                "--certificate-fingerprint",
```
Comment (erdembayar) [inline src/Tests/dotnet.Tests/CommandTests/CompleteCommandTests.cs:289]: "--certificate-fingerprint"  this one goes little bit up.

**Review by erdembayar — APPROVED** [review]
Small comments on order of parameters. But it doesn't block.

File: src/Tests/dotnet.Tests/CommandTests/CompleteCommandTests.cs (hunk 1 above)
Comment (heng-liu) [reply src/Tests/dotnet.Tests/CommandTests/CompleteCommandTests.cs:293]: Thanks. Fixed the order.
I think the sequence doesn't affect the testing results, as we will sort both the real results and the expected results.

File: src/Tests/dotnet.Tests/CommandTests/CompleteCommandTests.cs (hunk 2 above)
Comment (heng-liu) [reply src/Tests/dotnet.Tests/CommandTests/CompleteCommandTests.cs]: Thanks. Fixed the order.
I think the sequence doesn't affect the testing results, as we will sort both the real results and the expected results.

**Pull request merged**